mne/channels/_dig_montage_utils.py

```diff
@@ -0,0 +1,306 @@
+    hpi = list()
+    elp = list()
+    point_names = list()
+    fids = dict()
+    dig_ch_pos = dict()
+    for d in dig:
+        if d['kind'] == FIFF.FIFFV_POINT_CARDINAL:
+            _check_frame(d, 'head')
+            fids[_cardinal_ident_mapping[d['ident']]] = d['r']
+        elif d['kind'] == FIFF.FIFFV_POINT_HPI:
+            _check_frame(d, 'head')
+            hpi.append(d['r'])
+            elp.append(d['r'])
+            point_names.append('HPI%03d' % d['ident'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EXTRA:
+            _check_frame(d, 'head')
+            hsp.append(d['r'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EEG:
+            _check_frame(d, 'head')
+            dig_ch_pos['EEG%03d' % d['ident']] = d['r']
+
+    return Bunch(
+        nasion=fids['nasion'], lpa=fids['lpa'], rpa=fids['rpa'],
+        hsp=np.array(hsp) if len(hsp) else None,
+        hpi=hpi,  # XXX: why this is returned as empty list instead of None?
+        elp=np.array(elp) if len(elp) else None,
+        point_names=point_names,
+        dig_ch_pos=dig_ch_pos,
+        coord_frame='head',
+    )
+
+
+def _read_dig_montage_egi(
+        fname,
+        _scaling,
+        _all_data_kwargs_are_none,
+):
+
+    if not _all_data_kwargs_are_none:
+        raise ValueError('hsp, hpi, elp, point_names, fif must all be '
+                         'None if egi is not None')
+    _check_fname(fname, overwrite='read', must_exist=True)
+
+    root = ElementTree.parse(fname).getroot()
+    ns = root.tag[root.tag.index('{'):root.tag.index('}') + 1]
+    sensors = root.find('%ssensorLayout/%ssensors' % (ns, ns))
+    fids = dict()
+    dig_ch_pos = dict()
+
+    fid_name_map = {'Nasion': 'nasion',
+                    'Right periauricular point': 'rpa',
+                    'Left periauricular point': 'lpa'}
+
+    for s in sensors:
+        name, number, kind = s[0].text, int(s[1].text), int(s[2].text)
+        coordinates = np.array([float(s[3].text), float(s[4].text),
+                                float(s[5].text)])
+
+        coordinates *= _scaling
+
+        # EEG Channels
+        if kind == 0:
+            dig_ch_pos['EEG %03d' % number] = coordinates
+        # Reference
+        elif kind == 1:
+            dig_ch_pos['EEG %03d' %
+                       (len(dig_ch_pos.keys()) + 1)] = coordinates
+            # XXX: we should do something with this (ref and eeg get mixed)
+
+        # Fiducials
+        elif kind == 2:
+            fid_name = fid_name_map[name]
+            fids[fid_name] = coordinates
+        # Unknown
+        else:
+            warn('Unknown sensor type %s detected. Skipping sensor...'
+                 'Proceed with caution!' % kind)
+
+    return Bunch(
+        # EGI stuff
+        nasion=fids['nasion'], lpa=fids['lpa'], rpa=fids['rpa'],
+        dig_ch_pos=dig_ch_pos, coord_frame='unknown',
+        # not EGI stuff
+        hsp=None, hpi=None, elp=None, point_names=None,
+    )
+
+
+def _foo_get_data_from_dig(dig):
```

Comment:
I don't like the `_foo` naming, what does it add? why not just `_get_data_from_dig`?

Reply:
This should definitely change. I kept it `_foo`, because it was easier for me to track than `# XXX` when moving things around (even more when I start had to do multiple branches and PRs that converged at some point).